src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralEdge.java

```diff
@@ -108,5 +108,25 @@
 
         return Arrays.asList(query);
     }
+
+    @Override
+    public List<String> generateDeleteByIdScript(@NonNull GremlinSource source) {
+        if (!(source instanceof GremlinSourceEdge)) {
+            throw new GremlinUnexpectedSourceTypeException("should be the instance of GremlinSourceEdge");
+        }
+
+        final List<String> scriptList = new ArrayList<>();
+        final String id = source.getId();
+
+        Assert.notNull(id, "id should not be null");
+
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_GRAPH);
+        scriptList.add(String.format(Constants.GREMLIN_PRIMITIVE_EDGE, id));
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_DROP);
+
+        final String query = String.join(Constants.GREMLIN_PRIMITIVE_INVOKE, scriptList);
+
+        return Arrays.asList(query);
```

Comment:
Seems all the script is a single `String`, but all the methods in this class are wrapping the `single String` into a `String list`, any reason why not using `String` as return type here? Just curious.

Reply:
We can consider that all query is compose of string list. In this way, we can avoiding query string handling mess like joint/contract/replace. Just take care of its own.